JSON-Report-Mapping failed due to missing fileName property

**Describe the bug**

This is similar to issue #220 and some others, but for a different property. I have version 2.0.3 installed. Like the referenced issue, this is for a Node JS project, and it's failing in the same spot, trying to do the JSON-Report-Mapping step.

It is not happening 100% of the time, but it has happened every time since I updated to 2.0.3 (I can't say if failures before that were caused by this same issue, because I didn't have verbose logging on).

I see in the generated json file (attached) that there are plenty of fileName properties. The error points to the end of the file, so I don't know where it is complaining about.

```
2020-03-26T16:57:59.1935039Z 11:57:59.188 INFO: Process Dependency-Check report
2020-03-26T16:57:59.1943171Z 11:57:59.188 INFO: Using JSON-Reportparser
2020-03-26T16:57:59.6277059Z 11:57:59.625 WARN: JSON-Analysis aborted
2020-03-26T16:57:59.6315402Z 11:57:59.625 DEBUG: Problem with JSON-Report-Mapping
2020-03-26T16:57:59.6315888Z org.sonar.dependencycheck.parser.ReportParserException: Problem with JSON-Report-Mapping
2020-03-26T16:57:59.6316741Z 	at org.sonar.dependencycheck.parser.JsonReportParserHelper.parse(JsonReportParserHelper.java:44)
2020-03-26T16:57:59.6317236Z 	at org.sonar.dependencycheck.DependencyCheckSensor.parseAnalysis(DependencyCheckSensor.java:63)
2020-03-26T16:57:59.6317824Z 	at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:121)
2020-03-26T16:57:59.6318363Z 	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
2020-03-26T16:57:59.6318811Z 	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
2020-03-26T16:57:59.6319276Z 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:354)
2020-03-26T16:57:59.6319721Z 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
2020-03-26T16:57:59.6320168Z 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
2020-03-26T16:57:59.6320627Z 	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
2020-03-26T16:57:59.6321088Z 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
2020-03-26T16:57:59.6321728Z 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
2020-03-26T16:57:59.6322361Z 	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
2020-03-26T16:57:59.6322713Z 	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
2020-03-26T16:57:59.6323155Z 	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
2020-03-26T16:57:59.6323592Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-26T16:57:59.6324025Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-26T16:57:59.6324515Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-26T16:57:59.6325255Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-03-26T16:57:59.6325676Z 	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
2020-03-26T16:57:59.6326068Z 	at com.sun.proxy.$Proxy0.execute(Unknown Source)
2020-03-26T16:57:59.6326408Z 	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
2020-03-26T16:57:59.6326824Z 	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
2020-03-26T16:57:59.6327212Z 	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
2020-03-26T16:57:59.6327668Z 	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
2020-03-26T16:57:59.6328189Z 	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
2020-03-26T16:57:59.6328617Z Caused by: com.fasterxml.jackson.databind.JsonMappingException: Missing required creator property 'fileName' (index 0)
2020-03-26T16:57:59.6329297Z  at [Source: sun.nio.ch.ChannelInputStream@4a5e598e; line: 1, column: 7250832] (through reference chain: org.sonar.dependencycheck.parser.element.Analysis["dependencies"]->java.util.ArrayList[11562])
2020-03-26T16:57:59.6330225Z 	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270)
2020-03-26T16:57:59.6331230Z 	at com.fasterxml.jackson.databind.DeserializationContext.reportMappingException(DeserializationContext.java:1247)
2020-03-26T16:57:59.6331808Z 	at com.fasterxml.jackson.databind.deser.impl.PropertyValueBuffer._findMissing(PropertyValueBuffer.java:192)
2020-03-26T16:57:59.6332351Z 	at com.fasterxml.jackson.databind.deser.impl.PropertyValueBuffer.getParameters(PropertyValueBuffer.java:159)
2020-03-26T16:57:59.6332859Z 	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:229)
2020-03-26T16:57:59.6333377Z 	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:138)
2020-03-26T16:57:59.6333890Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:471)
2020-03-26T16:57:59.6334720Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1191)
2020-03-26T16:57:59.6335274Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314)
2020-03-26T16:57:59.6335783Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:183)
2020-03-26T16:57:59.6336260Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:150)
2020-03-26T16:57:59.6336772Z 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:287)
2020-03-26T16:57:59.6337339Z 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:259)
2020-03-26T16:57:59.6338224Z 	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:26)
2020-03-26T16:57:59.6338880Z 	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)
2020-03-26T16:57:59.6339640Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511)
2020-03-26T16:57:59.6340243Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:401)
2020-03-26T16:57:59.6340827Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1191)
2020-03-26T16:57:59.6341372Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314)
2020-03-26T16:57:59.6341873Z 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148)
2020-03-26T16:57:59.6342320Z 	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3814)
2020-03-26T16:57:59.6342747Z 	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2924)
2020-03-26T16:57:59.6343206Z 	at org.sonar.dependencycheck.parser.JsonReportParserHelper.parse(JsonReportParserHelper.java:40)
2020-03-26T16:57:59.6343538Z 	... 24 common frames omitted
```

**To Reproduce**

This is the json report file.
[dependency-check-report.zip](https://github.com/dependency-check/dependency-check-sonar-plugin/files/4388672/dependency-check-report.zip)

**Versions (please complete the following information):**

-   dependency-check: 5.2.1002 (Azure DevOps plug-in)
-   sonarqube: Enterprise 8.2.0.32929 
-   dependency-check-sonar-plugin: 2.0.3

It resolved it, but now I get a different error. Would you like me to file a new bug report for that?

[dependency-check-report (2).zip](https://github.com/dependency-check/dependency-check-sonar-plugin/files/4394120/dependency-check-report.2.zip)

```
2020-03-27T14:34:01.1561574Z ##[error]09:34:00.613 ERROR: Error during SonarQube Scanner execution
2020-03-27T14:34:01.1563183Z 09:34:00.613 ERROR: Error during SonarQube Scanner execution
2020-03-27T14:34:01.1563946Z ##[error]java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
2020-03-27T14:34:01.1565437Z java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
2020-03-27T14:34:01.1567630Z ##[error]at org.sonar.dependencycheck.reason.NPMDependencyReason.fillArtifactMatch(NPMDependencyReason.java:107)
	at org.sonar.dependencycheck.reason.NPMDependencyReason.getBestTextRange(NPMDependencyReason.java:88)
	at org.sonar.dependencycheck.reason.DependencyReason.isDependencyReasonBetterForDependencyThen(DependencyReason.java:119)
	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.getBestDependencyReason(DependencyReasonSearcher.java:104)
2020-03-27T14:34:01.1569724Z 	at org.sonar.dependencycheck.reason.NPMDependencyReason.fillArtifactMatch(NPMDependencyReason.java:107)
2020-03-27T14:34:01.1570283Z 	at org.sonar.dependencycheck.reason.NPMDependencyReason.getBestTextRange(NPMDependencyReason.java:88)
2020-03-27T14:34:01.1570838Z 	at org.sonar.dependencycheck.reason.DependencyReason.isDependencyReasonBetterForDependencyThen(DependencyReason.java:119)
2020-03-27T14:34:01.1572452Z ##[error]at org.sonar.dependencycheck.reason.DependencyReasonSearcher.linkIssuesToDependencyReasons(DependencyReasonSearcher.java:169)
	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.addDependenciesToInputComponents(DependencyReasonSearcher.java:138)
2020-03-27T14:34:01.1573949Z 
2020-03-27T14:34:01.1574340Z 	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.linkIssuesToDependencyReasons(DependencyReasonSearcher.java:169)
2020-03-27T14:34:01.1574952Z 	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.addDependenciesToInputComponents(DependencyReasonSearcher.java:138)
2020-03-27T14:34:01.1582715Z ##[error]at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:153)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:354)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
```

@Reamer I am getting the error with that build. Is there someplace specifically in the log I should look?

```
2020-03-28T21:16:36.0637511Z 16:16:34.118 DEBUG: No Identifier with type javascript found for Dependency yeast:0.1.2
2020-03-28T21:16:36.0637861Z 16:16:34.118 DEBUG: Found a name and version match in Eclipse_Backend/package-lock.json
2020-03-28T21:16:36.0638243Z 16:16:34.118 DEBUG: DependencyReasonWinner: Eclipse_Backend/package-lock.json
2020-03-28T21:16:36.0638584Z 16:16:34.118 DEBUG: Get the best DependencyReason out of 1 for moment
2020-03-28T21:16:36.0638901Z 16:16:34.118 DEBUG: No Identifier with type javascript found for Dependency moment
2020-03-28T21:16:36.0639222Z 16:16:34.118 WARN: Optional[moment] doesn't have a "@". Checkout dependency moment
2020-03-28T21:16:36.0639515Z 16:16:34.118 DEBUG: eslint-bridge server will shutdown
2020-03-28T21:16:36.0639864Z 16:16:35.431 DEBUG: stylelint-bridge server will shutdown
2020-03-28T21:16:36.0640181Z 16:16:35.447 INFO: ------------------------------------------------------------------------
2020-03-28T21:16:36.0640452Z 16:16:35.447 INFO: EXECUTION FAILURE
2020-03-28T21:16:36.0640736Z 16:16:35.447 INFO: ------------------------------------------------------------------------
2020-03-28T21:16:36.0641018Z 16:16:35.447 INFO: Total time: 8:44.305s
2020-03-28T21:16:36.0641246Z 16:16:35.478 INFO: Final Memory: 32M/1536M
2020-03-28T21:16:36.0641546Z 16:16:35.478 INFO: ------------------------------------------------------------------------
2020-03-28T21:16:36.0643039Z ##[error]16:16:35.478 ERROR: Error during SonarQube Scanner execution
2020-03-28T21:16:36.0644444Z 16:16:35.478 ERROR: Error during SonarQube Scanner execution
2020-03-28T21:16:36.0645061Z ##[error]java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
2020-03-28T21:16:36.0646182Z java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
2020-03-28T21:16:36.0647662Z ##[error]at org.sonar.dependencycheck.reason.NPMDependencyReason.fillArtifactMatch(NPMDependencyReason.java:110)
	at org.sonar.dependencycheck.reason.NPMDependencyReason.getBestTextRange(NPMDependencyReason.java:88)
	at org.sonar.dependencycheck.reason.DependencyReason.isDependencyReasonBetterForDependencyThen(DependencyReason.java:119)
2020-03-28T21:16:36.0649273Z 	at org.sonar.dependencycheck.reason.NPMDependencyReason.fillArtifactMatch(NPMDependencyReason.java:110)
2020-03-28T21:16:36.0649715Z 	at org.sonar.dependencycheck.reason.NPMDependencyReason.getBestTextRange(NPMDependencyReason.java:88)
2020-03-28T21:16:36.0650192Z 	at org.sonar.dependencycheck.reason.DependencyReason.isDependencyReasonBetterForDependencyThen(DependencyReason.java:119)
2020-03-28T21:16:36.0656531Z ##[error]at org.sonar.dependencycheck.reason.DependencyReasonSearcher.getBestDependencyReason(DependencyReasonSearcher.java:104)
	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.linkIssuesToDependencyReasons(DependencyReasonSearcher.java:169)
	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.addDependenciesToInputComponents(DependencyReasonSearcher.java:138)
	at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:153)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:354)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-03-28T21:16:36.0660734Z 	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.getBestDependencyReason(DependencyReasonSearcher.java:104)
2020-03-28T21:16:36.0661242Z 	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.linkIssuesToDependencyReasons(DependencyReasonSearcher.java:169)
2020-03-28T21:16:36.0661757Z 	at org.sonar.dependencycheck.reason.DependencyReasonSearcher.addDependenciesToInputComponents(DependencyReasonSearcher.java:138)
2020-03-28T21:16:36.0662207Z 	at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:153)
2020-03-28T21:16:36.0662609Z 	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
2020-03-28T21:16:36.0662991Z 	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
2020-03-28T21:16:36.0663393Z 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:354)
2020-03-28T21:16:36.0663862Z 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
2020-03-28T21:16:36.0664249Z 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
2020-03-28T21:16:36.0664614Z 	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
2020-03-28T21:16:36.0665003Z 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
2020-03-28T21:16:36.0665375Z 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
2020-03-28T21:16:36.0665712Z 	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
2020-03-28T21:16:36.0666015Z 	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
2020-03-28T21:16:36.0666379Z 	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
2020-03-28T21:16:36.0666766Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-03-28T21:16:36.0667139Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-03-28T21:16:36.0667595Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-03-28T21:16:36.0667955Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-03-28T21:16:36.0670367Z ##[error]at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
```
[dependency-check-report (3).zip](https://github.com/dependency-check/dependency-check-sonar-plugin/files/4397927/dependency-check-report.3.zip)

--- a/sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/parser/element/Identifier.java
+++ b/sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/parser/element/Identifier.java
@@ -60,6 +60,7 @@ public class Identifier {
             // pkg:maven/struts/struts@1.2.8 -> maven
             // pkg:javascript/jquery@2.2.0 -> javascript
             // pkg:npm/arr-flatten@1.1.0 -> npm
+            // pkg:npm/mime -> npm
             return Optional.of(StringUtils.substringAfter(StringUtils.substringBefore(identifier.getId(), "/"), "pkg:"));
         }
         return Optional.empty();
@@ -69,6 +70,7 @@ public class Identifier {
             // pkg:maven/struts/struts@1.2.8 -> struts/struts@1.2.8
             // pkg:javascript/jquery@2.2.0 -> jquery@2.2.0
             // pkg:npm/arr-flatten@1.1.0 -> arr-flatten@1.1.0
+            // pkg:npm/mime -> mime
             return Optional.of(StringUtils.substringAfter(identifier.getId(), "/"));
         }
         return Optional.empty();

--- a/sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/reason/NPMDependencyReason.java
+++ b/sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/reason/NPMDependencyReason.java
@@ -41,6 +41,7 @@ import org.sonar.dependencycheck.reason.npm.PackageLockModel;
 
 import edu.umd.cs.findbugs.annotations.CheckForNull;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 
 public class NPMDependencyReason extends DependencyReason {
 
@@ -99,12 +100,22 @@ public class NPMDependencyReason extends DependencyReason {
     }
 
     private void fillArtifactMatch(@NonNull Dependency dependency, Identifier npmIdentifier) {
-        Optional<String> packageArtifact = Identifier.getPackageArtifact(npmIdentifier);
-        if (packageArtifact.isPresent()) {
-            // packageArtifact has something like jquery@2.2.0
-            String[] npmIdentifierSplit = packageArtifact.get().split("@");
-            String name = npmIdentifierSplit[0];
-            String version = npmIdentifierSplit[1];
+        String packageArtifact = Identifier.getPackageArtifact(npmIdentifier).orElse(null);
+        if (StringUtils.isNotBlank(packageArtifact)) {
+            String name;
+            String version;
+            if (packageArtifact.contains("@")) {
+                // packageArtifact is something like jquery@2.2.0
+                String[] npmIdentifierSplit = packageArtifact.split("@");
+                name = npmIdentifierSplit[0];
+                version = npmIdentifierSplit[1];
+            } else {
+                // It happens, that packageArtifact doesn't contain a version
+                // https://github.com/dependency-check/dependency-check-sonar-plugin/issues/242#issuecomment-605521827
+                name = packageArtifact;
+                version = null;
+            }
+
             // Try to find in <dependency>
             for (NPMDependency npmDependency : packageLockModel.getDependencies()) {
                 checkNPMDependency(name, version , npmDependency)
@@ -113,7 +124,7 @@ public class NPMDependencyReason extends DependencyReason {
         }
     }
 
-    private Optional<TextRangeConfidence> checkNPMDependency(String name, String version, NPMDependency dependency) {
+    private Optional<TextRangeConfidence> checkNPMDependency(String name, @Nullable String version, NPMDependency dependency) {
         if (StringUtils.equals(name, dependency.getName())
                 && StringUtils.equals(version, dependency.getVersion())) {
             LOGGER.debug("Found a name and version match in {}", packageLock);

--- a/sonar-dependency-check-plugin/src/test/java/org/sonar/dependencycheck/parser/element/IdentifierTest.java
+++ b/sonar-dependency-check-plugin/src/test/java/org/sonar/dependencycheck/parser/element/IdentifierTest.java
@@ -48,6 +48,16 @@ class IdentifierTest {
     }
 
     @Test
+    public void testNodeWithOutVersion() {
+        Identifier a = new Identifier("pkg:npm/mime", Confidence.HIGHEST);
+        assertFalse(Identifier.isMavenPackage(a));
+        assertTrue(Identifier.isNPMPackage(a));
+        assertFalse(Identifier.isJavaScriptPackage(a));
+        assertEquals("mime", Identifier.getPackageArtifact(a).get());
+        assertEquals("npm", Identifier.getPackageType(a).get());
+    }
+
+    @Test
     public void testJavaScript() {
         Identifier a = new Identifier("pkg:javascript/jquery@2.2.0", Confidence.HIGHEST);
         assertFalse(Identifier.isMavenPackage(a));

--- a/sonar-dependency-check-plugin/src/test/java/org/sonar/dependencycheck/reason/NPMDependencyReasonTest.java
+++ b/sonar-dependency-check-plugin/src/test/java/org/sonar/dependencycheck/reason/NPMDependencyReasonTest.java
@@ -127,6 +127,26 @@ public class NPMDependencyReasonTest extends DependencyReasonTestHelper {
     }
 
     @Test
+    public void foundDependencyNPMWithoutVersion() throws IOException {
+        NPMDependencyReason npm = new NPMDependencyReason(inputFile("package-lock.json"));
+        // Create Dependency
+        Identifier identifier = new Identifier("pkg:npm/arr-flatten", Confidence.HIGHEST);
+        Collection<Identifier> identifiersCollected = new ArrayList<>();
+        identifiersCollected.add(identifier);
+        Dependency dependency = new Dependency(null, null, null, null, Collections.emptyMap(),Collections.emptyList(), identifiersCollected, Collections.emptyList(), null);
+        TextRangeConfidence textRangeConfidence = npm.getBestTextRange(dependency);
+        assertTrue(npm.isReasonable());
+        assertNotNull(textRangeConfidence);
+        assertEquals(7, textRangeConfidence.getTextrange().start().line());
+        assertEquals(0, textRangeConfidence.getTextrange().start().lineOffset());
+        assertEquals(11, textRangeConfidence.getTextrange().end().line());
+        assertEquals(6, textRangeConfidence.getTextrange().end().lineOffset());
+        assertEquals(Confidence.HIGH, textRangeConfidence.getConfidence());
+        // verify that same dependency points to the same TextRange, use of HashMap
+        assertEquals(npm.getBestTextRange(dependency), npm.getBestTextRange(dependency));
+    }
+
+    @Test
     public void foundNoDependency() throws IOException {
         NPMDependencyReason npm = new NPMDependencyReason(inputFile("package-lock.json"));
         // Create Dependency

